src/material/schematics/ng-generate/navigation/schema.json

```diff
@@ -92,6 +92,11 @@
       "type": "boolean",
       "default": false,
       "description": "Specifies if the component is an entry component of declaring module."
+    },
+    "routing": {
```

Comment:
I personally think that the schematic should be doing this check based on the app module and not provided as an option.

Reply:
That might be overly complicated as we'd need to check the imports of the module. These checks are not reliable most of the time either as in reality the Angular compiler supports partial evaluation (i.e. module can be specified in a separate file through an exported variable).

Surely we could use the static interpreter as in FW update migrations, but as said, that might be out of scope.